### PR TITLE
mat64: return Condition(inf) from Dense.Inverse for singular matrix

### LINKED ISSUE #410: matrix/mat64: Dense.Inverse does not fail on singular matrix
One should expect Dense.Inverse to return an error for singular matrices. Instead, it returns garbage: for a 2x2, it returns the input matrix, for a 4x4 it returns the input matrix where some values are changed. I have not checked for other sizes.

Code:
```go
package main

import (
	"fmt"

	"github.com/gonum/matrix/mat64"
)

func main() {
	A := mat64.NewDense(2, 2, []float64{0, 0, 0, 1})
	var InvA mat64.Dense
	err := InvA.Inverse(A)
	if err != nil {
		panic(err)
	}
	fmt.Printf("%v\n", mat64.Formatted(&InvA, mat64.Prefix("")))

	B := mat64.NewDense(4, 4, []float64{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 20, 20, 0, 0, 20, 20})
	var InvB mat64.Dense
	err = InvB.Inverse(B)
	if err != nil {
		panic(err)
	}
	fmt.Printf("%v\n", mat64.Formatted(&InvB, mat64.Prefix("")))
}
```

Result:
```
⎡0  0⎤
⎣0  1⎦
⎡ 0   0   0   0⎤
⎢ 0   0   0   0⎥
⎢ 0   0  20  20⎥
⎣ 0   0   1   0⎦
```

### PR DESCRIPTION
**Pull request by vladimir-ch**
Fixes #410

### CODE REVIEW

**Comment by kortschak** [comment]
Can we have a test for this too.

**Comment by vladimir-ch** [comment]
I will add it.

**Comment by kortschak** [comment]
Looking at the coverage, there are parts that I am troubled with and are not covered. I'll file an issue for improving the tests on Inverse.

File: mat64/dense_arithmetic.go [hunk 1]
```diff
@@ -239,7 +239,10 @@ func (m *Dense) Inverse(a Matrix) error {
 		}
 	}
 	ipiv := make([]int, r)
-	lapack64.Getrf(m.mat, ipiv)
+	ok := lapack64.Getrf(m.mat, ipiv)
```
Comment (btracey) [inline mat64/dense_arithmetic.go]: Oops. Sorry.

**Comment by vladimir-ch** [comment]
Ha, another bug in Dense.Inverse discovered. I've fixed it and added the tests with singular matrix. PTAL

**Pull request merged**